src/include/deque.h

```diff
@@ -280,6 +280,13 @@ pgmoneta_deque_to_string(struct deque* deque, int32_t format, char* tag, int ind
 void
 pgmoneta_deque_destroy(struct deque* deque);
 
+/**
+ * Set the deque to be thread safe
+ * @param deque The deque
+ */
+void
+pgmoneta_deque_set_thread_safe(struct deque* deque);
```

Comment:
Why do we need this ? its _create(true, ...

Reply:
Yeah it's a little ad-hoc. The deque is inside json so it's not created with thread safe set and can only be set later. We can revisit this if we have more use cases as such.